File: backend/src/routes/tools.js
import { Router } from "express";

import { asyncHandler, validateBody } from "../middleware.js";
import { emiCalculationSchema } from "../schemas.js";
import { calculateEmi } from "../services/recommendationService.js";

export const toolsRouter = Router();

toolsRouter.post(
  "/debug/calculate-emi",
  validateBody(emiCalculationSchema),
  asyncHandler(async (req, res) => {
    const { amount, interest_rate, months } = req.validated;
    res.json(calculateEmi(amount, interest_rate, months));
  }),
);

File: backend/app/api/routes/tools.py
from fastapi import APIRouter

from app.schemas.tool import EMICalculationRequest, EMICalculationResponse
from app.services.recommendation_service import calculate_emi

router = APIRouter(tags=["tools"])


@router.post("/debug/calculate-emi", response_model=EMICalculationResponse)
def calculate_emi_route(payload: EMICalculationRequest) -> EMICalculationResponse:
    result = calculate_emi(
        amount=payload.amount,
        annual_interest_rate=payload.interest_rate,
        months=payload.months,
    )
    return EMICalculationResponse(**result)
